fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -551,8 +554,29 @@ public boolean visit(java.lang.String elementName, int elementIndex, Quantity qu
         }
         if (quantity.getValue() != null && quantity.getValue().hasValue()) {
             BigDecimal value = quantity.getValue().getValue();
-            BigDecimal valueLow = NumberParmBehaviorUtil.generateLowerBound(value);
-            BigDecimal valueHigh = NumberParmBehaviorUtil.generateUpperBound(value);
+            BigDecimal valueLow = null;
+            BigDecimal valueHigh = null;
+            if (quantity.getComparator() != null) {
+                // Calculate range based on comparator
+                if (QuantityComparator.GREATER_OR_EQUALS.equals(quantity.getComparator())) {
+                    // Range low value is quantity value
+                    valueLow = value;
+                } else if (QuantityComparator.GREATER_THAN.equals(quantity.getComparator())) {
+                    // Range low value is quantity value plus 1e-<scale+9>
+                    int scale = value.scale() < 0 ? 9 : value.scale()+9;
+                    valueLow = value.add(new BigDecimal("1e-" + scale));
+                } else if (QuantityComparator.LESS_OR_EQUALS.equals(quantity.getComparator())) {
+                    // Range high value is quantity value
+                    valueHigh = value;
+                } else if (QuantityComparator.LESS_THAN.equals(quantity.getComparator())) {
+                    // Range high value is quantity value minus 1e-<scale+9>
+                    int scale = value.scale() < 0 ? 9 : value.scale()+9;
+                    valueHigh = value.subtract(new BigDecimal("1e-" + scale));
+                }
```

Comment:
Isn't the Else If actually an Else? (e.g. it wasn't the previous 3 so it must be an else.  It'd skip a comparison.

Reply:
changed to just Else